sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -610,10 +611,24 @@ applying the annotations when the object is stored to memory.
 a|
 [source,c++]
 ----
-annotated_ref& operator=(const annotated_ref&) = default;
+T operator=(const annotated_ref&);
 ----
 |
-Assign from another `annotated_ref` object.
+Writes an object of type `T` to the location referenced by this wrapper,
+applying the annotations when the object is stored to memory.
+Does not rebind the reference!
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+T operatorOP(const T &);
+----
+|
+Where [code]#OP# is: [code]#pass:[+=]#, [code]#-=#,[code]#*=#, [code]#/=#, [code]#%=#, [code]#+<<=+#, [code]#>>=#, [code]#&=#, [code]#\|=#, [code]#^=#.
+
+Compound assignment operators. Return result by value.
+Available if the corresponding non-compound operator is available for `T`.
```

Comment:
I think it is good if the API can be implemented both with the attribute approach (our current implementation) and built-in load/store. For the later it isn't possible to implement it with `+=` because it needs to be something like `store(p, load(p)+v)`. For any reasonable type it doesn't matter because `+=` and `+` should do the same. But I specified it to be `+` (and changed the implementation to use it) so that is the same between our implementation and a hypothetical implementation using built-in load/store.